### PR TITLE
Feat/basic settings page

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [ ] Chore
- [x] Feature
- [ ] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

https://digicatapult.atlassian.net/browse/VR-248

## High level description

Allows user to view the settings page.

## Detailed description

Settings page with a reset button and the only editable field being the admin email. 

https://github.com/user-attachments/assets/0a24a2b8-22f5-47e2-88b2-4839fbb29e1e

Showing correct address and company name on Bob:

https://github.com/user-attachments/assets/55737720-eb97-4371-9943-1dff5a66108b

If `DEMO_MODE` is false:

https://github.com/user-attachments/assets/97617b0e-106a-4316-bad4-96af3f2515c5

## Describe alternatives you've considered

N/A

### CODE REVIEW

File: src/controllers/settings/index.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    return this.html(this.settingsTemplates.settings(set))
+  }
+
+  /**
+   * handles PIN code submission form submit action
+   * @param body - contains forms inputs
+   * @returns
+   */
+  @SuccessResponse(200)
+  @Post('/update/:edit?')
```
Comment (mattdean-digicatapult) [inline src/controllers/settings/index.ts:62]: I'd recommend this just be `/` rather than introducing a verb in the path

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
This looks really really good. Just a few small changes

**Comment by jonmattgray** [comment]
Coming to this unfamiliar with the repo so please ignore if I'm wrong. In the video, I'd expect the settings icon in the sidebar to be highlighted when you're on the settings page, it highlights the top icon instead?

File: test/e2e/helpers/setupConnection.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  if (host === null || port === null) {
+    throw new Error(`Unspecified smtp4dev host or port ${smtp4devUrl}`)
+  }
+  const { inviteEmail, adminEmail } = await checkEmails(host, port)
+  const extractedPin = await extractPin(adminEmail.id, smtp4devUrl)
+  if (!extractedPin) throw new Error('PIN for Bob was not found.')
+  if (extractedPin.length !== 6) {
+    throw new Error('Pin does not have the expected length.')
+  }
+  const pinForBob = extractedPin
```
Comment (jonmattgray) [inline test/e2e/helpers/setupConnection.ts:24]: can delete?

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**